server/service/osquery_utils/queries.go

```diff
@@ -890,7 +894,22 @@ SELECT
   '' AS vendor,
   0 AS last_opened_at,
   path AS installed_path
-FROM homebrew_packages;
+FROM homebrew_packages
+WHERE type = 'formula'
+UNION
+SELECT
+  name AS name,
+  version AS version,
+  '' AS bundle_identifier,
+  '' AS extension_id,
+  '' AS browser,
+  'homebrew_packages' AS source,
+  '' AS vendor,
+  0 AS last_opened_at,
+  path AS installed_path
+FROM homebrew_packages
+WHERE type = 'cask'
+AND NOT EXISTS (SELECT 1 FROM file WHERE file.path LIKE CONCAT(homebrew_packages.path, '/%%%%') AND file.path LIKE '%%.app%%' LIMIT 1);
```

Comment:
After some testing I found out what's happening behind the scenes:
```sql
SELECT 1 FROM file 
WHERE file.path LIKE CONCAT(homebrew_packages.path, '/%%')
AND file.path LIKE '%.app' LIMIT 1
```

1. osquery will first walk the first directory in the `LIKE`: `CONCAT(homebrew_packages.path, '/%%')` and generate the list of paths.
2. osquery will then walk the other directory in the `LIKE`: '%.app', which given it's not an absolute path will try to look for `.app`s in the current working directory (which for osqueryd usually run as `root` will be `/`) and generate the list of paths, which on macOS will be an empty list.

So, only (1) will generate paths to process.

After (1) and (2), the sqlite engine will perform its own filtering with `LIKE` (string matching) over the returned paths in (1).

So, basically `LIKE` is used first by osquery code to generate the paths AND then by the sqlite engine which will do string matching as usual.

The trick here is that we can use this because we are expecting `/` to contain no apps and no large number of files or directories (I was not able to create files or `.app`s in `/`).

Reply:
Rather than relying on "no large number of files or directories" to make this query efficient, should we instead use `filename LIKE '%.app'`?